### ISSUE TITLE
bug: PyCharm - incorrect import suggestions for schemas

### ISSUE THREAD

**Issue by Tomas2D** [IBM/ibm-generative-ai#279]
Currently, the `PyCharm` suggests importing re-exported schemas from an incorrect module.

For instance: `TextGenerationParamters` should be imported from `genai.text.generation`, but PyCharm suggests importing it from the `genai._generated.api` which is not desired.

**Comment by jezekra1** [comment]
## The proposed solution:

1. **Unify all schemas into a single file, instead of distributed imports per service:**
Instead of `from genai.text.generation import DecodingMethod` -> `from genai.schema import DecodingMethod`.
2. **Structure of the schemas**:
```
genai
  |- schemas
     |- __init__.py
        _api.py
        _endpoints.py
        _extensions.py 
``` 
- `_api.py` will be divided to public and private schemas (prefixed with `_`) already in the generator
- `__init__.py`:
    - will export all public schemas from `_api.py` (`from _api import *` hides private classes by default)
    - will export everything from `_extensions.py` which allow defining classes not generated (e.g `HumanMessage`)
    - can be used to modify a schema -> (`import MySchema from _api as MySchema_`, modify and reexport under original name)
3. **Add public schemas to generator config and modify behavior to make all private by default**
4. **rewrite all examples and documentation to use `from genai.schema` imports**
5. **modify sevice-local schemas**:
- `.schema` files (`genai/text/generation/schema.py`) will now just be an alias for `genai.schema`
- show deprecation warning: "Use `from genai.schema import ...` instead of `genai.text.generation.schema`
    - should be possible using ["module-level" getattr](https://peps.python.org/pep-0562/)

**Comment by Owaiskhan9654** [comment]
Is this a known issue in `ibm-generative-ai==2.0.0`? I updated this package and using the new migration code. 
@Tomas2D @jezekra1 

<img width="1189" alt="image" src="https://github.com/IBM/ibm-generative-ai/assets/47840160/3fff6fee-cfe8-44ff-bfb7-b563707263b5">

**Comment by jezekra1** [comment]
The version containing these changes hasn't been released yet, you're probably looking at the unreleased version of the documentation.

Please use the old migration guide for now or wait till the next release

https://ibm.github.io/ibm-generative-ai/v2.0.0/v2_migration_guide.html

**Comment by Tomas2D** [comment]
Released in v2.1.0
